p2p/src/conn.rs

```diff
@@ -322,8 +330,11 @@ where
 		.name("peer_write".to_string())
 		.spawn(move || {
 			let mut retry_send = Err(());
+			writer
+				.set_write_timeout(Some(BODY_IO_TIMEOUT))
+				.expect("set timeout");
 			loop {
-				let maybe_data = retry_send.or_else(|_| send_rx.recv_timeout(IO_TIMEOUT));
+				let maybe_data = retry_send.or_else(|_| send_rx.recv_timeout(HEADER_IO_TIMEOUT));
```

Comment:
This is actually reading a msg from our internal msg queue, not the tcp stream.
So I think we should keep a specific timeout for this.

Reply:
done